numberguesser/src/Form.js

```diff
@@ -0,0 +1,40 @@
+import React, { useState, useEffect } from "react";
+
+const Button = (props) => {
+  return (
+    <button onClick={function () {
+      props.setIfClicked(state => (true));
+      console.log('actual',props.actualAnswer)
+      console.log('input',props.inputNumber)
+
+      if (props.actualAnswer == props.inputNumber) {
+        props.setCompareAnswer(state => (true))
+      } else {
+        props.setCompareAnswer(state => (false))
+      }
+    }
+    }
+    >Check!</button>
+  )
+}
+
+export default function Form(props) {
+  return (
+    <div className="input-your-guess">
+      <form autoComplete="off" onSubmit={event => event.preventDefault()}>
+        <input
+          className="guess-number"
+          type="number"
+          placeholder="Guess a number!"
+          onChange={(event) => {
+            props.setInputNumber(event.target.value)
+          }
+          }
+          value={props.inputNumber}
+          data-testid="number-input"
+        />
+      </form>
+      <Button setIfClicked={props.setIfClicked} inputNumber={props.inputNumber} actualAnswer={props.actualAnswer} setCompareAnswer={props.setCompareAnswer}/>
```

Comment:
Having this button changing a lot of states it is a bit confusing to follow through. It would've been cleaner and easier to read this if those changes were done through a function set on App.js and pass it through this component as a prop.